app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
```

Comment:
Hum I didn't think of that :/ 

You can reload the statuses only at lest with `statuses.reload`.